web/src/shared/components/annotator/components/text-annotation/helpers.ts

```diff
@@ -89,8 +89,18 @@ export const updateCurrentTextAnnBound = (curBound: Bound, token: PageToken): Bo
     };
 };
 
-export const isTokenOnNewLine = (token: PageToken, curBound: Bound): boolean => {
-    return isBoundValid(curBound) && token.y > curBound.y + token.height;
+const LINE_BREAK_CHAR = '\n';
+const isTokenLastOfLine = (token: PageToken) => token?.text!.includes(LINE_BREAK_CHAR);
```

Comment:
Is "?" and "!" needed here? As I see based on the type should be defined here or there are some cases when something goes wrong?

Reply:
When prevToken is declared, it is declared as an empty object and casted to Bound. During the first check, it could cause problems so I used both.